### PR TITLE
Remove travel schedule from registration

### PR DESCRIPTION
**Pull request by bear454**
(cherry picked from commit d18cc02f0185df39bb4213d8a63a955dfb433dcb)
Re: https://github.com/openSUSE/osem/issues/2333

**Checklist**

- [x] I have read the [Contribution & Best practices Guide](https://github.com/openSUSE/osem/blob/master/CONTRIBUTING.md).
- [x] My branch is up-to-date with the upstream `master` branch.
- [x] The tests pass locally with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works(if appropriate).

**Short description of what this resolves/which [issues](https://github.com/openSUSE/osem/issues) does this fix?:**

Get rid of travel info on registrations.

- #2333 

**Changes proposed in this pull request:**

- Remove travel info from registration views & database.

### CODE REVIEW

File: spec/controllers/conference_registration_controller_spec.rb [hunk 1]
```diff
@@ -290,13 +290,12 @@
...
                                conference: conference,
-                               user:       user,
-                               arrival:    Date.new(2014, 04, 25))
+                               user:       user)
       end
 
       context 'updates successfully' do
         before do
-          patch :update, params: { registration:  attributes_for(:registration, arrival: Date.new(2014, 04, 29)),
+          patch :update, params: { registration:  attributes_for(:registration),
```
Comment (Ana06) [inline spec/controllers/conference_registration_controller_spec.rb]: I think we should either give it other parameters or drop this test. This is not really testing something.

File: spec/factories/registration.rb [hunk 2]
```diff
@@ -6,5 +6,6 @@
   factory :registration do
     user
     conference
+    other_special_needs { '' }
```
Comment (Ana06) [inline spec/factories/registration.rb]: unrelated

File: app/views/conference_registrations/_form.html.haml [hunk 3]
```diff
@@ -22,6 +22,7 @@
             = render partial: 'devise/shared/sign_up_form_embedded'
 
             = render partial: 'registration_info', locals: { f: f }
+            = f.input :conference_id, as: :hidden, value: @conference.id
```
[the remark this answers is not in the document]
Comment (bear454) [reply app/views/conference_registrations/_form.html.haml]: It's not, but AFAICT it's missing from the form, and just plain broken without it.

File: app/controllers/surveys_controller.rb [hunk 4]
```diff
@@ -40,6 +40,11 @@ def reply
       end
     end
 
-    redirect_back(fallback_location: root_path)
+    case @survey.target
```
[the remark this answers is not in the document]
Comment (bear454) [reply app/controllers/surveys_controller.rb]: Unrelated fix; removed in eb6319a8.

File: db/schema.rb [hunk 5]
```diff
@@ -10,10 +10,10 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20181113195810) do
+ActiveRecord::Schema.define(version: 20181229233811) do
 
   create_table "answers", force: :cascade do |t|
-    t.string "title"
+    t.string   "title"
```
[the remark this answers is not in the document]
Comment (bear454) [reply db/schema.rb]: Cleaned up in eb6319a8.

File: spec/controllers/conference_registration_controller_spec.rb (hunk 1 above)
Comment (bear454) [reply spec/controllers/conference_registration_controller_spec.rb]: Added a different (random) parameter.

File: spec/factories/registration.rb (hunk 2 above)
Comment (bear454) [reply spec/factories/registration.rb]: Whoops; removed in eb6319a8.

**Comment by bear454** [comment]
Squashed for convenience; rebased to clear the Hakiri warning.

File: db/migrate/20181229233811_remove_arrival_departure_from_registrations.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+class RemoveArrivalDepartureFromRegistrations < ActiveRecord::Migration[5.0]
+  def change
+    remove_column :registrations, :arrival, :datetime
```
Comment (differentreality) [inline db/migrate/20181229233811_remove_arrival_departure_from_registrations.rb]: So do we plan to just remove this information, without migrating it to surveys?

Comment (bear454) [reply db/migrate/20181229233811_remove_arrival_departure_from_registrations.rb]: For my case, at LFNW, over ~2700 registrations on 2 events, we have 0 values recorded for travel info.

I don't see any value in writing a translation for that.

Comment (bear454) [reply db/migrate/20181229233811_remove_arrival_departure_from_registrations.rb]: In general, I don't agree with keeping the old data indefinitely anyway (and I think the GDPR agrees on this point).

One can always create a survey for a _future_ conference that collects the data... but what is the use case for keeping the old travel data?

Comment (Ana06) [reply db/migrate/20181229233811_remove_arrival_departure_from_registrations.rb]: I honestly don't think anybody is using this information. I think it is fine to just removing without migrating it. We should though add this to the [CHANGES file](https://github.com/openSUSE/osem/blob/master/CHANGES.md) (we tend to forget to document changes :see_no_evil:)

File: app/views/conference_registrations/_form.html.haml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ana06) [reply app/views/conference_registrations/_form.html.haml]: Would you mind putting it in a different commit? I think it will be helpful in the future to separate those two (someone may want for example to pick the commit with the bug fix without removing the travel information or we may debug this code in a year and wonder how is this related).

[the remark this answers is not in the document]
Comment (bear454) [reply app/views/conference_registrations/_form.html.haml]: I've removed that line from the commit; it's not _required_ in upstream `osem` at this time, but I do recall now why I need it, and why it's a good practice.

Upstream, the conference is being identified using routing parameters. AT LFNW, we're not.

In general, using routing parameters in form data creates a non-obvious dependency of any form submission on the routing table; e.g. changing a route can cause a form submission to fail, unexpectedly. I find it to be more obvious, and a clear separation of responsibilities, to simply include all necessary attributes in the form data; routing is therefore limited to the responsibility of (surprise) routing, and the form data is complete for form submission, and neither affects the other.

File: db/migrate/20181229233811_remove_arrival_departure_from_registrations.rb (hunk 6 above)
Comment (bear454) [reply db/migrate/20181229233811_remove_arrival_departure_from_registrations.rb]: Added a note to CHANGES in bddfcd6.

**Comment by Ana06** [comment]
It seems something went wrong while rebasing. Tests are failing and the commit has a strange message:

```
[branch remove-travel-schedule] Remove travel schedule from registration
(cherry picked from commit d18cc02f0185df39bb4213d8a63a955dfb433dcb)
Re: #2333
```

Other than that, looks good :+1:

File: app/views/conference_registrations/_form.html.haml (hunk 3 above)
[the remark this answers is not in the document]
Comment (bear454) [reply app/views/conference_registrations/_form.html.haml]: > It seems something went wrong while rebasing. Tests are failing and the commit has a strange message

The tests require that form element, for context :doh:

[the remark this answers is not in the document]
Comment (bear454) [reply app/views/conference_registrations/_form.html.haml]: I restored the form element, which _should_ allow the strong params to work again.

**Comment by differentreality** [comment]
Shall we do this? :)

**Comment by bear454** [comment]
> Shall we do this? :)

Not without an approving review ;-)

**Review by hennevogel — APPROVED** [review]
